cirq-core/cirq/sim/simulation_state.py

```diff
@@ -166,6 +166,37 @@ def create_merged_state(self) -> Self:
         """Creates a final merged state."""
         return self
 
+    def add_qubits(self: Self, qubits: Sequence['cirq.Qid']) -> Self:
```

Comment:
I think it's a bit confusing that the new functions return something when they do their operations inplace. lets `-> None` . since it always returns `self`. if it creates new state then it's still the same `self` object but with new paramters.